fluentd/lib/filter_elasticsearch_genid_ext/lib/filter_elasticsearch_genid_ext.rb

```diff
@@ -2,35 +2,85 @@
 require 'base64'
 require 'fluent/filter'
 
+begin
+  ViaqMatchClass = Fluent::Match
+rescue
+  # Fluent::Match not provided with 0.14
+  class ViaqMatchClass
+    def initialize(pattern_str, unused)
+      patterns = pattern_str.split(/\s+/).map {|str|
+        Fluent::MatchPattern.create(str)
+      }
+      if patterns.length == 1
+        @pattern = patterns[0]
+      else
+        @pattern = Fluent::OrMatchPattern.new(patterns)
+      end
+    end
+    def match(tag)
+      @pattern.match(tag)
+    end
+    def to_s
+      "#{@pattern}"
+    end
+  end
+end
+
 module Fluent
   class ElasticsearchGenidExtFilter < Filter
     Fluent::Plugin.register_filter('elasticsearch_genid_ext', self)
 
+    desc 'key to store generated unique id or the value of alt_key if specified'
     config_param :hash_id_key, :string, :default => '_hash'
+    desc 'key for the hash "record" (optional)'
     config_param :alt_key, :string, default: ""
+    desc 'process alt_key in records with this tag pattern'
+    config_param :alt_tags, :array, default: []
+
+    def configure(conf)
+      super
+      @alt_keys = @alt_key.split('.')
+      if ENV['GENID_ALT_TAG']
```

Comment:
By using `alt_tags "#{ENV['GENID_ALT_TAG'] || 'tag patterns string'}"` in the config, you can get rid of this